lib/features/booking/widgets/update_phone_no_bottom_sheet_view.dart

```diff
@@ -112,11 +180,82 @@ class _UpdatePhoneNoBottomSheetBody
                             top: HsDimens.spacing8,
                             left: HsDimens.spacing16,
                             right: HsDimens.spacing16),
-                        child: PrimaryButton(
-                          label: HatSpaceStrings.current.save,
-                          // TODO: validate On Press event
+                        child: ValueListenableBuilder<PhoneNumberErrorType?>(
+                          valueListenable: _phoneNumberError,
+                          builder: (context, value, child) {
+                            return PrimaryButton(
+                                label: HatSpaceStrings.current.save,
+                                onPressed: _phoneNumber == null
+                                    ? null
+                                    : value?.phoneNumberError != null
+                                        ? null
+                                        : () {
+                                            // TODO: SAVE AND UPLOAD TO DATABASE
+                                          });
+                          },
                         )))
               ],
             )));
   }
 }
+
+class PhoneNumberInputFormatter extends TextInputFormatter {
+  final ValueNotifier<PhoneNumberErrorType?> error;
+
+  PhoneNumberInputFormatter(this.error);
+  @override
+  TextEditingValue formatEditUpdate(
+      TextEditingValue oldValue, TextEditingValue newValue) {
+    // FORMAT PHONE NUMBER
+    if (oldValue.text.length < newValue.text.length) {
+      if (newValue.text.length == 5) {
+        return TextEditingValue(
+            text:
+                '${oldValue.text.substring(0, 4)} ${newValue.text.substring(newValue.text.length - 1)}');
```

Comment:
What format is this?

Reply:
The format of input text must be XXXX XXX XXX so I came up with this approach.